server/src/test/java/com/objectcomputing/checkins/services/memberskill/MemberSkillControllerTest.java

```diff
@@ -0,0 +1,224 @@
+        final HttpResponse<Set<MemberSkill>> response = client.toBlocking().exchange(request, Argument.setOf(MemberSkill.class));
+
+        assertEquals(memberSkills, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+        verify(memberSkillServices, times(1)).findByFields(null,null);
+    }
+
+    @Test
+    void testReadMemberSkill() {
+        MemberSkill m = new MemberSkill(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
+
+        when(memberSkillServices.read(eq(m.getId()))).thenReturn(m);
+
+        final HttpRequest<Object> request = HttpRequest.GET(String.format("/%s", m.getId().toString())).basicAuth("MEMBER", "MEMBER");
+        final HttpResponse<MemberSkill> response = client.toBlocking().exchange(request, MemberSkill.class);
+
+        assertEquals(m, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+        verify(memberSkillServices, times(1)).read(any(UUID.class));
+    }
+
+    @Test
+    void testReadMemberSkillNotFound() {
+        MemberSkill m = new MemberSkill(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
+
+        when(memberSkillServices.read(eq(m.getId()))).thenReturn(null);
+
+        final HttpRequest<Object> request = HttpRequest.GET(String.format("/%s", m.getId().toString())).basicAuth("MEMBER", "MEMBER");
+        HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class, () -> client.toBlocking().exchange(request, MemberSkill.class));
+
+        assertEquals(HttpStatus.NOT_FOUND, responseException.getStatus());
+
+        verify(memberSkillServices, times(1)).read(any(UUID.class));
+    }
+
+    @Test
+    void testFindMemberSkills() {
+        MemberSkill m = new MemberSkill(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
+        Set<MemberSkill> memberSkillSet = Collections.singleton(m);
+
+        when(memberSkillServices.findByFields(eq(m.getMemberid()), eq(m.getSkillid()))).thenReturn(memberSkillSet);
+
+        final HttpRequest<?> request = HttpRequest.GET(String.format("/?memberid=%s&skillid=%s", m.getMemberid(),
+                m.getSkillid())).basicAuth("MEMBER", "MEMBER");
+        final HttpResponse<Set<MemberSkill>> response = client.toBlocking().exchange(request, Argument.setOf(MemberSkill.class));
+
+        assertEquals(memberSkillSet, response.body());
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+        verify(memberSkillServices, times(1)).findByFields(any(UUID.class), any(UUID.class));
+    }
+
+    @Test
+    void testFindMemberSkillsNull() {
+        MemberSkill m = new MemberSkill(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
+
+        when(memberSkillServices.findByFields(eq(m.getMemberid()), eq(null))).thenReturn(null);
+
+        final HttpRequest<?> request = HttpRequest.GET(String.format("/?memberid=%s", m.getMemberid())).basicAuth("MEMBER", "MEMBER");
```

Comment:
Don't hardcode 'MEMBER' use the constants please. Replace this everywhere you write it out as a literal.

Reply:
👍